### DISCUSSION TITLE
sync with postgres on demand, with sync with users still on

### DISCUSSION THREAD

**Discussion by fredericrous** [electric-sql/electric#649] [Feature requests]
I see scenarios where we need sync between users but we don't need postgres on cloud all the time. Let me give some examples:

- Let say I create a video game. My user scores a point, I want to sync that on the server. When my user move, I want the other users to see he moved, but I do not need to store these moves on postgres
- Now I'm creating a workflow builder for a no code app. My user can collaborate with an expert or with his team on creating the workflow. when he is happy. He uses the publish button, sync with postgres happens
- Another one: a chat application where users can choose not to store their history
- Last one: a team composed of many operators that can control a robot. I don't need to store on the cloud all instructions the robot receives, I just need to store when an unexpected issue occurs, and info needed for audit like when did it start moving, when did it complete its task

I went through the docs, and a few videos. I don't think this is possible yet? I see "Shapes" on the roadmap but it does not seem to answer this problem?

**Comment by thruflo** [comment]
Yes, at the moment everything syncs via the database (Postgres).

We can add ephemeral sync that just goes via the sync service. For example to support presence, and the scenarios you illustrate above. This is technically possible and something we’re aware that multi-user apps will benefit from.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
